rotkehlchen/chain/ethereum/modules/makerdao/sai/decoder.py

```diff
@@ -0,0 +1,521 @@
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+        if tx_log.topics[0] == MAKERDAO_SAITUB_WITHDRAW_TOPIC:
+            return self._decode_makerdao_withdraw_eth(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+
+        return None, []
+
+    def _decode_makerdao_withdraw_eth(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: list[HistoryBaseEntry],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[list[ActionItem]],  # pylint: disable=unused-argument
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        cdp_id = hex_or_bytes_to_int(tx_log.topics[2])
+        amount_withdrawn_raw = hex_or_bytes_to_int(tx_log.topics[3])
+        amount_withdrawn = asset_normalized_value(amount=amount_withdrawn_raw, asset=self.eth)
+
+        for event in decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount.is_close(amount_withdrawn, max_diff='0.003') and
+                event.asset == self.eth
+            ):
+                event.event_type = HistoryEventType.WITHDRAWAL
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET
+                event.counterparty = CPT_SAI
+                event.notes = f'Withdraw {amount_withdrawn} ETH from CDP {cdp_id}'
+
+        return None, []
+
+    def _decode_makerdao_new_cdp_event(
+            self,
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: list[HistoryBaseEntry],  # pylint: disable=unused-argument
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[list[ActionItem]],  # pylint: disable=unused-argument
+    ) -> tuple[Optional[HistoryBaseEntry], list[ActionItem]]:
+        cdp_creator = hex_or_bytes_to_address(tx_log.topics[1])
+        cdp_id = hex_or_bytes_to_int(tx_log.data[:32])
+        event = HistoryBaseEntry(
+            event_identifier=transaction.tx_hash,
+            timestamp=ts_sec_to_ms(transaction.timestamp),
+            sequence_index=self.base_tools.get_next_sequence_counter(),
+            location=Location.BLOCKCHAIN,
+            asset=self.eth,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            balance=Balance(),
+            location_label=cdp_creator,
+            counterparty=CPT_SAI,
+            notes=f'Create CDP {cdp_id}',
```

Comment:
Keep in mind these events have `extra_data`, on which you can add anything you want such as the `cdp_id` if you want to use it and compare with other events in the same transactions. If not then don't use it to take up space without a reason, but know you have this option.